Assets/HoloToolkit/Input/Scripts/Gaze/GazeManager.cs

```diff
@@ -0,0 +1,349 @@
+
+        /// <summary>
+        /// The LayerMasks, in prioritized order, that are used to determine the HitObject when raycasting.
+        ///
+        /// Example Usage:
+        ///
+        /// // Allow the cursor to hit SR, but first prioritize any DefaultRaycastLayers (potentially behind SR)
+        ///
+        /// int sr = LayerMask.GetMask("SR");
+        /// int nonSR = Physics.DefaultRaycastLayers & ~sr;
+        /// GazeManager.Instance.RaycastLayerMasks = new LayerMask[] { nonSR, sr };
+        /// </summary>
+        [Tooltip("The LayerMasks, in prioritized order, that are used to determine the HitObject when raycasting.\n\nExample Usage:\n\n// Allow the cursor to hit SR, but first prioritize any DefaultRaycastLayers (potentially behind SR)\n\nint sr = LayerMask.GetMask(\"SR\");\nint nonSR = Physics.DefaultRaycastLayers & ~sr;\nGazeManager.Instance.RaycastLayerMasks = new LayerMask[] { nonSR, sr };")]
+        public LayerMask[] RaycastLayerMasks = new LayerMask[] { Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Current stabilization method, used to smooth out the gaze ray data.
+        /// If left null, no stabilization will be performed.
+        /// </summary>
+        [Tooltip("Stabilizer, if any, used to smooth out the gaze ray data.")]
+        public BaseRayStabilizer Stabilizer = null;
+
+        /// <summary>
+        /// Transform that should be used as the source of the gaze position and orientation.
+        /// Defaults to the main camera.
+        /// </summary>
+        [Tooltip("Transform that should be used to represent the gaze position and orientation. Defaults to Camera.Main")]
+        public Transform GazeTransform;
+
+        /// <summary>
+        /// Dispatched when focus shifts to a new object, or focus on current object
+        /// is lost.
+        /// </summary>
+        public event FocusedChangedDelegate FocusedObjectChanged;
+
+        private float lastHitDistance = 2.0f;
+
+        /// <summary>
+        /// Unity UI pointer event.  This will be null if the EventSystem is not defined in the scene.
+        /// </summary>
+        public PointerEventData UnityUIPointerEvent { get; private set; }
+
+        protected override void Awake()
+        {
+            base.Awake();
+
+            // Add default RaycastLayers as first layerPriority
+            if (RaycastLayerMasks == null || RaycastLayerMasks.Length == 0)
+            {
+                RaycastLayerMasks = new LayerMask[] { Physics.DefaultRaycastLayers };
+            }
+
+            if (GazeTransform == null)
+            {
+                if (Camera.main != null)
+                {
+                    GazeTransform = Camera.main.transform;
+                }
+                else
+                {
+                    Debug.LogError("Gaze Manager was not given a GazeTransform and no main camera exists to default to.");
+                }
+            }
+        }
+
+        private void Update()
```

Comment:
I agree with splitting the code in functions. From what I remember, it wasn't trivial to do because of dependencies between some variables, but can take a quick stab at it tomorrow.

I'll look at the focus point stuff as well. Might just be a matter of adding it to the InputManager prefab.